cmake/modules/configuration_files.cmake

```diff
@@ -77,8 +77,15 @@ zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards DTS APP_BOARD_DTS SUFFIX
 
 zephyr_get(DTC_OVERLAY_FILE SYSBUILD LOCAL)
 if(NOT DEFINED DTC_OVERLAY_FILE)
-  zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR} DTS DTC_OVERLAY_FILE
-              NAMES "${APP_BOARD_DTS};${BOARD}.overlay;app.overlay" SUFFIX ${FILE_SUFFIX})
+  foreach(file ${APP_BOARD_DTS})
+    zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR} DTS DTC_OVERLAY_FILE
+                NAMES "${file}" SUFFIX ${FILE_SUFFIX})
```

Comment:
Suffix is already applied here: https://github.com/zephyrproject-rtos/zephyr/blob/719b3380555e68843efa54073a256dd62c231751/cmake/modules/configuration_files.cmake#L76

So I assume this code will duplicate `SUFFIX` when used.

Reply:
Noticed other issues from this, have applied fixes